extensions/wikia/CommunityPage/scripts/firstEditModal.js

```diff
@@ -0,0 +1,130 @@
+require([
+	'jquery',
+	'wikia.ui.factory',
+	'wikia.mustache',
+	'communitypage.templates.mustache',
+	'wikia.nirvana',
+	'wikia.tracker',
+	'wikia.window',
+	'wikia.cookies',
+], function ($, uiFactory, mustache, templates, nirvana, tracker, window, cookies) {
+	'use strict';
+
+	 var track = tracker.buildTrackingFunction({
+		 action: tracker.ACTIONS.CLICK,
+		 category: 'community-page-first-edit',
+		 trackingMethod: 'analytics'
+	 });
+
+	function getUiModalInstance() {
+		var $deferred = $.Deferred();
+
+		uiFactory.init(['modal']).then(function (uiModal) {
+			$deferred.resolve(uiModal);
+		});
+
+		return $deferred;
+	}
+
+	function getModalContents() {
+		var $deferred = $.Deferred();
+
+		nirvana.sendRequest({
+			controller: 'CommunityPageSpecial',
+			method: 'getFirstTimeEditorModalData',
+			data: {
+				uselang: window.wgUserLanguage
+			},
+			format: 'json',
+			type: 'get',
+		}).then(function (response) {
+			$deferred.resolve(response);
+		});
+
+		return $deferred;
+	}
+
+	function openModal() {
+		// Track impression
+		track({
+			action: tracker.ACTIONS.IMPRESSION,
+			label: 'first-edit-modal-loaded',
+		});
+
+		$.when(
+			getUiModalInstance(),
+			getModalContents()
+		).then(function (uiModal, modalContents) {
+			var createPageModalConfig = {
+				vars: {
+					classes: ['CommunityPageFirstEditModal'],
+					content: '',
+					id: 'CommunityPageFirstEditModal',
+					size: 'medium',
+				}
+			};
+
+			uiModal.createComponent(createPageModalConfig, function (modal) {
+				modal.$content
+					.html(mustache.render(templates.firstEditModal, {
+						heading: modalContents.headingText,
+						subheading: modalContents.subheadingText,
+						getStarted: modalContents.getStartedText,
+						maybeLater: modalContents.maybeLaterText,
+						getStartedLink: modalContents.getStartedLink,
+					}));
+
+				modal.show();
+				initModalTracking(modal);
```

Comment:
`initModalTracking` currently doesn't use a parameter but it is called with one here. Did you mean to pass the modal (or even `modal.$element`)?